services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/allocator/IpAddrAllocator.java

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.privateipmanager.allocator;
+
+
+import com.futurewei.alcor.privateipmanager.exception.IpAddrNotEnoughException;
+
+import java.util.List;
+
+public interface IpAddrAllocator {
+
+    String allocate() throws Exception;
+
+    List<String> allocateBulk(int num) throws IpAddrNotEnoughException, Exception;
+
+    void release(String ipAddr) throws Exception;
+
+    void releaseBulk(List<String> ipAddrList) throws Exception;
+
+    boolean valid(String ipv4Addr);
```

Comment:
validate?

Reply:
This method checks whether the  ipv4Addr is valid in this ip range, which means greater than or equal to firstAddr and less than or equal to lastAddr.

Reply:
Correct. I meant to use the verb "validate" as the method name to be consistent with other names.